test/conftest.py

```diff
@@ -52,7 +52,18 @@ def toxiproxy_server():
         ['toxiproxy-server', '-port', str(port), '-host', host],
         stdout=subprocess.PIPE
     )
-    time.sleep(0.2)  # allow server to start
+
+    class NotReady(Exception):
+        pass
+
+    @retry(delay=0.1, max_attempts=10)
```

Comment:
Actually no, we want to catch exceptions from requests here too, e.g. if the server isn't accepting connections yet.